clients/java/src/main/java/io/camunda/zeebe/client/api/worker/JobWorkerMetrics.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.zeebe.client.api.worker;
+
+import io.camunda.zeebe.client.api.worker.metrics.MicrometerJobWorkerMetricsBuilder;
+import io.camunda.zeebe.client.impl.worker.metrics.MicrometerJobWorkerMetricsBuilderImpl;
+
+/** Worker metrics API. Allows basic instrumenting of job activation and handling. */
+public interface JobWorkerMetrics {
+
+  /**
+   * Called every time one or more jobs are activated.
+   *
+   * <p>NOTE: this is called <em>before</em> the job is worked on.
+   *
+   * @param count the amount of jobs that were activated
+   */
+  default void jobActivated(final int count) {}
```

Comment:
I opted to make all methods default to simplify partial implementations, i.e. not all users may care about all metrics :shrug: